designs/sam-debugging/local-sam-debugging.md

```diff
@@ -70,66 +68,59 @@ SAM CLI related parameters affect how the application is built and launched. For
 
-Two CodeLenses are added: "Run Locally", and "Debug Locally". The only difference between the two is whether or not a debugger is involved. When users click either CodeLens, the Toolkit shows a selection picker. The picker presents users with the following choices:
-
--   "Invoke \<Debug Configuration Name\>" - The picker contains one of these for every debug configuration found that references this SAM Template and resource. If users select one of these choices, the toolkit launches the corresponding debug configuration, as if the user launched it from VS Code's Debug view.
--   "Add Debug Configuration" - If users select this choice the toolkit produces an `aws-sam` Debug Configuration that is configured to invoke the resource being acted on, and adds it to `launch.json`. Users are taken to the new entry in `launch.json` instead of starting a debug session.
+The Toolkit adds an "Add Debug Configuration" CodeLens above resources that are not already referenced by a debug configuration. When users click this CodeLens, the toolkit produces a Debug Configuration of type `aws-sam`, configures it to invoke the resource, and adds it to `launch.json`. Users are taken to the new entry in `launch.json`.
 
 #### CodeLenses in Code files
 
-The Toolkit adds CodeLenses to functions considered [eligible Lambda handlers](#eligible-lambda-handler).
-
-Two CodeLenses are added: "Run Locally", and "Debug Locally". The only difference between the two is whether or not a debugger is involved. When users click either CodeLens, the Toolkit shows a selection picker. The picker presents users with the following choices:
+The Toolkit offers a way for users to set up debug configurations that can launch functions considered [eligible Lambda handlers](#eligible-lambda-handler).
 
--   "Invoke \<Debug Configuration Name\>" - The picker contains one of these for every debug configuration found that references this lambda handler. This includes debug configurations that directly reference this function, and those that reference this function through SAM Template resources. If users select one of these choices, the toolkit launches the corresponding debug configuration, as if the user launched it from VS Code's Debug view.
--   "Add Debug Configuration" - If users select this choice the toolkit produces an `aws-sam` Debug Configuration that is configured to invoke the resource being acted on, and adds it to `launch.json`. Users are taken to the new entry in `launch.json` instead of starting a debug session.
+The Toolkit adds an "Add Debug Configuration" CodeLens above lambda handers that are not already referenced by a debug configuration. When users click this CodeLens, the toolkit produces a Debug Configuration of type `aws-sam`, configures it to invoke the lambda handler, and adds it to `launch.json`. Users are taken to the new entry in `launch.json`.
```

Comment:
(section also applies to `CodeLenses in SAM Template files`)
>The Toolkit adds an "Add Debug Configuration" CodeLens above lambda handers that are not already referenced by a debug configuration

Two questions with the above: is this detection mechanism something we can feasibly implement, and do we think it's not valuable to persist the codelens so a user can create a second config without copy/paste?

Also, could you provide a note (either here or in the appendix) with the information we can feasibly autopopulate for a user? I imagine we're not going to give them a blank JSON blob, correct?